Pa3/Durchführung/Project/app/StateHandler.c

```diff
@@ -31,13 +31,190 @@
 /* PROTOTYPES *************************************************************************************/
 
 /* VARIABLES **************************************************************************************/
-static EventEnum gCurrentState;
+static EventEnum gCurrentEvent = NO_EVENT_HAS_HAPPEND; 
+static StateEnum gCurrentState = STATE_INIZALIZATION_MCU; //First state when the MCU is started
 static SoftTimer* gLapTimer;
+static bool gEntryDone = false;
 
 /* EXTERNAL FUNCTIONS *****************************************************************************/
 
 void StateHandler_process(void)
 {
+  switch(gCurrentState)
+  { 
+    case STATE_INIZALIZATION_MCU: 
+      InitializeMcu_InitializeAll(void); //entry
+      InitializeMcu_DisplayTeamName(void); //exit
+      gCurrentState = STATE_CALIBRATE_LINE_SENSORS;
+      break;
+
+    case STATE_CALIBRATE_LINE_SENSORS:
+      if(gEntryDone == false)
+      {
+        gEntryDone = true;
+        gCurrentEvent = CalibrateLineSensors_Initialize(void); //entry
+        if (NO_EVENT_HAS_HAPPEND != gCurrentEvent)
+        {
+          gCurrentState = STATE_ERROR_HANDLER;
+        }
+      }
+
+      gCurrentEvent = CalibrateLineSensors_CalibrateSensors(void); //do
+      switch (gCurrentEvent)
+      {
+        case CALIBRATION_FAILED:
+          gCurrentState = STATE_ERROR_HANDLER;
+          break;
+
+        case CALIBRATION_DONE:
+          gEntryDone = false;
+          gCurrentState = STATE_READY_TO_DRIVE;
+          break;
+
+        default:
+          break; 
+      }
+      break;
+
+    case STATE_READY_TO_DRIVE:         
+      gCurrentEvent = ReadyToDrive_CheckStateOfButtons(void); //do
+
+      switch(gCurrentEvent)
+      {
+        case CALIBRATION_BUTTON_HAS_BEEN_RELEASED:
+          gCurrentState = STATE_CALIBRATE_LINE_SENSORS;
+          break;
+
+        case START_BUTTON_HAS_BEEN_RELEASED:
+          gCurrentState = STATE_DISPLAY_COUNTDOWN;
+          break;
+
+        case PARAM_BUTTON_HAS_BEEN_RELEASED:
+          gCurrentState = STATE_SET_PARAMETERS;
+          break;
+
+        default:
+          break; 
+      }
+      break;
+
+    case STATE_SET_PARAMETERS
+      SetParameters_SetNextParamSet(void); //entry
+      SetParameters_DisplayParamSet(void); //exit
+      gCurrentState = STATE_READY_TO_DRIVE;
+      break;
+
+    case STATE_DISPLAY_COUNTDOWN:      
+      if(gEntryDone == false)
+      {
+        gEntryDone = true;
+        DisplayCountdown_StartCountdown(void); //entry
+      }
+      gCurrentEvent = DisplayCountdown_DisplayCountdown(void); //do
+
+      if (gCurrentEvent == COUNTDOWN_IS_FINISHED)
```

Comment:
please check all comparisons for yoda
```suggestion
      if (COUNTDOWN_IS_FINISHED ==gCurrentEvent)
```